libraries/redcore/api/hal/hal.php

```diff
@@ -1374,7 +1374,7 @@ public function checkRequiredFields($data, $configuration)
 			{
 				if (RApiHalHelper::isAttributeTrue($field, 'isRequiredField'))
 				{
-					if (empty($data[(string) $field['name']]))
+					if (is_null($data[(string) $field['name']]))
```

Comment:
we should check for (is_null($data[(string) $field['name']]) || $data[(string) $field['name']] == '') here not null, since we assign the default value or at least '' to it...